### PR TITLE
Added clearMemoryCache to driver

### LINKED ISSUE #719: Clear browser cache?
I'm running into intermittent test failures that look like the following:
```
     Failure/Error: expect(status_code).to eql 200
       
       expected: 200
            got: 304
```
Clearly what's happening is, dependent on the order of my tests, it appears sometimes the same URL has already been fetched in a previous test. This is causing Poltergeist/ Phantom is sending an `ETag` header, causing the server to respond with a 304 instead of a 200.  Note I already have `config.before( :each) { Capybara.reset_sessions! }`.

Now, obv I can change my tests but the fact remains that tests are not isolated due to browser caching.  I found the [Browser#reset_cache!](http://www.rubydoc.info/github/jnicklas/capybara/Capybara%2FRackTest%2FBrowser%3Areset_cache!) method in the Capybara API but clearly doesn't apply to the Poltergeist driver.  I see methods for clearing cookies and localStorage, but nothing to clear the browser cache.  Is this possible?

### PR DESCRIPTION
**Pull request by pgaertig**
This PR fixes #719 and replaces #756. To clear memory cache of PhantomJS just invoke in your Capybara tests:

     page.driver.clear_memory_cache

Use of this routine may be also workaround of the problem described in #754

### CODE REVIEW

**Comment by pgaertig** [comment]
Travis build fails because only PhantomJS 1.9.8 is used and `clearMemoryCache` function was introduced with PhantomJS 2.0.0. Locally the test passed with 2.0.0 and 2.1.1. Any suggestions how to best handle PhantomJS versioning in Poltergeist specs?

**Comment by twalpole** [comment]
@pgaertig Travis tests phantomjs 1.9.8, 2.0, and 2.1.1  -  You can use `phantom.version.major` to access the phantomjs version in the coffeescript, and return an error about lack of support if the version is < 2

**Comment by pgaertig** [comment]
Thanks @twalpole for suggestion. I will add function existence checking which I think is better than hard-binding to engine version. `clearMemoryCache` is not listed in public PhantomJS API docs yet and who knows. Regarding tests I see Travis is downloading 2.0.0 & 2.1.1 however there is line printed out with `master` tests:
   $ ${PHANTOMJS:-phantomjs} --version
   1.9.8
So I think 2.x is not user for `master`. I saw you your #744. Do you have any plans with PhantomJS' version aware conditional tests? Or should I just skip that test until Poltergeist `master` drop 1.9.x support?

**Comment by twalpole** [comment]
@pgaertig  the version printed by that command is the version of the system installed phantomjs, and not really relevant for the test run - the ones that are downloaded are used in some of the test runs where `env: PHANTOMJS=...` is specified - see [.travis.yml include matrix](https://github.com/teampoltergeist/poltergeist/blob/master/.travis.yml#L31)

I don't believe there are any current version aware tests, and the ruby side of the code doesn't really know what version of phantomjs is being used - you could add tests that only execute on travis by checking for ENV['TRAVIS'] and then if ENV['PHANTOMJS'] is set it would imply phantomjs >= 2.0

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@
 /rspec.failures
 /vendor
 /.bundle
+.idea
```
Comment (route) [inline .gitignore]: This is optional to your system. Could you please remove it?

**Comment by pgaertig** [comment]
Unfortunately I cannot commit to this PR anymore, #767 is new attempt with requested change.

**Pull request closed without merging**